src/Symfony/Component/Serializer/Stream/InputStream.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Stream;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental in 7.0
+ */
+final class InputStream extends Stream
```

Comment:
Is all this boilerplate code mandatory? Cannot we use the `resource` type directly? Or just provide a genetic class taking the name of a resource as parameter?